core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -221,19 +224,18 @@ protected Attributes getNameResolverParams() {
 
   private static class AuthorityOverridingTransportFactory implements ClientTransportFactory {
     final ClientTransportFactory factory;
-    @Nullable final String authorityOverride;
+    final String authorityOverride;
 
     AuthorityOverridingTransportFactory(
-        ClientTransportFactory factory, @Nullable String authorityOverride) {
+        ClientTransportFactory factory, String authorityOverride) {
```

Comment:
Nit: I know you didn't cause this, but would you mind calling checkNotNull on factory too?  Neither param can be null, but your change hints that factory could be null.